src/util/schnorr.rs

```diff
@@ -18,64 +18,114 @@
 //!
 
 pub use secp256k1::schnorrsig::{PublicKey, KeyPair};
-use secp256k1::{Secp256k1, Verification};
+use secp256k1::{Secp256k1, Verification, Error, constants, Signing};
 use hashes::{Hash, HashEngine};
 use util::taproot::{TapBranchHash, TapTweakHash};
+use core::{fmt, str};
 
 /// Untweaked Schnorr public key
 pub type UntweakedPublicKey = PublicKey;
 
 /// Tweaked Schnorr public key
+#[derive(Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Debug)]
 pub struct TweakedPublicKey(PublicKey);
 
+impl fmt::LowerHex for TweakedPublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::LowerHex::fmt(&self.0, f)
+    }
+}
```

Comment:
Any reason not to implement upper hex?

Reply:
Done

Reply:
It was absent since inner type does not implement it as well - and doing it should be a case of other PR. Here we just need to get the stuff needed for Taproot :)